src/main/java/uk/gov/hmcts/befta/DefaultBeftaTestDataLoader.java

```diff
@@ -0,0 +1,129 @@
+
+    private boolean isTestDataLoadedForCurrentRound = false;
+
+    @Override
+    public synchronized boolean isTestDataLoadedForCurrentRound() {
+        return this.isTestDataLoadedForCurrentRound;
+    }
+
+    @Override
+    public synchronized void loadTestDataIfNecessary() {
+        if (!isTestDataLoadedForCurrentRound && !shouldSkipDataLoad()) {
+            try {
+                doLoadTestData();
+                updateDataLoadDetailsInRecentExecutionsInfo();
+            } catch (Exception e) {
+                throw e;
+            } finally {
+                isTestDataLoadedForCurrentRound = true;
+            }
+        }
+    }
+
+    protected void doLoadTestData() {
+    }
+
+    private boolean shouldSkipDataLoad() {
+        try {
+            double testDataLoadSkipPeriod = BeftaMain.getConfig().getTestDataLoadSkipPeriod();
+            RecentExecutionsInfo recentExecutionsInfo = JsonUtils
+                    .readObjectFromJsonFile(TestAutomationAdapter.EXECUTION_INFO_FILE, RecentExecutionsInfo.class);
+            String recentExecutionTime = recentExecutionsInfo.getLastExecutionTime();
+
+            if (isWithinSkipPeriod(
+                    recentExecutionTime,
+                    testDataLoadSkipPeriod)
+                    && wasMostRecentDataForSameTests(recentExecutionsInfo)) {
+                return true;
+            }
+        } catch (Exception e) {
+            BeftaUtils.defaultLog("Should NOT skip loading data.", e);
+            return false;
+        }
+        return false;
+    }
+
+    private void updateDataLoadDetailsInRecentExecutionsInfo() {
+        String recentExecutionsInfoFilePath = TestAutomationAdapter.EXECUTION_INFO_FILE;
+        String dateTimeFormat = BeftaUtils.getDateTimeFormatRequested("now");
+        String currentTime = LocalDateTime.now().format(DateTimeFormatter.ofPattern(dateTimeFormat));
+        RecentExecutionsInfo recentExecutionsInfo = new RecentExecutionsInfo();
+        recentExecutionsInfo.setLastExecutionTime(currentTime);
+        recentExecutionsInfo.setLastExecutionProjectRepo(getCurrentGitRepo());
+        recentExecutionsInfo.setLastExecutionProjectBranch(getCurrentGitBranch());
+        try {
+            JsonUtils.writeJsonToFile(recentExecutionsInfoFilePath, recentExecutionsInfo);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private String getCurrentGitBranch() {
+        String branchName = "";
+        try {
+            Process process = Runtime.getRuntime().exec("git rev-parse --abbrev-ref HEAD");
+            process.waitFor();
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            branchName = reader.readLine();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return branchName;
+    }
+
+    private String getCurrentGitRepo() {
+        String repoName = "";
```

Comment:
why not user current directory name by default? this way, even if we get an exception below, we will have a nice name.